packages/react-native-reanimated/metro-config/index.js

```diff
@@ -0,0 +1,35 @@
+const COLLAPSED_STACK_REGEX = new RegExp(
+  [
+    // For internal usage in the example app
+    '/packages/react-native-reanimated/.+\\.(t|j)sx?$',
```

Comment:
Internal packages also acquire it via linked `node_modules`, is it necessary?

Reply:
Not quite. The path I am getting from metro is pointing to `packages/react-native-reanimated` instead of `node_modules/react-native-reanimated`.

<img width="492" alt="Screenshot 2024-08-12 at 15 56 03" src="https://github.com/user-attachments/assets/cba77e32-cef1-4641-896d-31de1a7f7ecb">

What is more, there is no `react-native-reanimated` symbolic link in `node_modules` in example apps so library sources are imported just from the `packages/react-native-reanimated`.

<img width="280" alt="Screenshot 2024-08-12 at 15 58 05" src="https://github.com/user-attachments/assets/ff18183d-643f-4abd-93b2-fbb9c602b6fe">

Reply:
The link is not in the `node_modules` of the package, but in `node_modules` of the whole monorepo.

![Screenshot 2024-08-13 at 09 50 46](https://github.com/user-attachments/assets/ea30e638-a46f-4e0a-bb3d-66a9a49c43f0)


I guess metro simply resolves this symbolic link to the actual path. How about instead of hardcoding the path in here we add an option to add ignored paths to the wrapper?

Reply:
Or maybe we should just forget whether it's `packages` or `node_modules` and just look for `react-native-reanimated`

Reply:
> Or maybe we should just forget whether it's packages or node_modules and just look for react-native-reanimated

No, it won't work as well. We won't be able to correct stack traces in our reanimated monorepo where the root directory is named `react-native-reanimated`. See the path `.../react-native-reanimated/packages/react-native-reanimated/...` in the screenshot above.

> I guess metro simply resolves this symbolic link to the actual path. How about instead of hardcoding the path in here we add an option to add ignored paths to the wrapper?

It might be useful but the benefit of exporting our metro-config wrapper will fade away. The user still can add custom logic to filter out stack frames via the `customizeFrame` method of the `symbolicator`.

I would prefer to keep the current metro config. If the user has a custom path to `react-native-reanimated` (rather a rare case, usually all packages leave in the `node_modules` directory), they can create a custom metro-config entry to filter out unwanted stack frames instead of using the provided wrappeer. I would rather add information to docs how it works and how to properly create such a config.

Reply:
Alternatively, we can add an option to provide a path to `node_modules` (but it seems unnecessary to me as metro in its default config assumes that `react-native` sources live under the `node_modules` directory, so I think we can do something similar).